main/src/cgeo/geocaching/models/Geocache.java

```diff
@@ -1398,12 +1399,20 @@ private void assignUniquePrefix(final Waypoint waypoint) {
             assignedPrefixes.add(wp.getPrefix());
         }
 
-        for (int i = OWN_WP_PREFIX_OFFSET; i < 100; i++) {
-            final String prefixCandidate = String.valueOf(i);
-            if (!assignedPrefixes.contains(prefixCandidate)) {
-                waypoint.setPrefix(prefixCandidate);
-                return;
+        // start with "17"
+        int a = 0;
+        int b = 6;
+        while (a < WP_PREFIX_CHARS.length()) {
```

Comment:
I would prefer to keep the style of the original loop. Rough idea:
```
final int length = WP_PREFIX_CHARS.length();
for (int i = OWN_WP_PREFIX_OFFSET; i < length * length; i++) {
final int a = i / length;
final int b = i % length;
...
```

Reply:
Interesting, most of the time, when I'm unsure with some lines of code, someone will comment on it :wink: 
I'll try your idea, thanks.

Still wondering where this `OWN_WP_PREFIX_OFFSET = 17` comes from. Does anyone know?